src/components/Swipeable.tsx

```diff
@@ -1,69 +1,104 @@
+import { State } from '../State';
 
 const DRAG_TOSS = 0.05;
 
-export type PropType = {
-  children: any,
-  friction: number,
-  leftThreshold?: number,
-  rightThreshold?: number,
-  overshootLeft?: boolean,
-  overshootRight?: boolean,
-  overshootFriction: number,
-  onSwipeableLeftOpen?: Function,
-  onSwipeableRightOpen?: Function,
-  onSwipeableOpen?: Function,
-  onSwipeableClose?: Function,
-  onSwipeableLeftWillOpen?: Function,
-  onSwipeableRightWillOpen?: Function,
-  onSwipeableWillOpen?: Function,
-  onSwipeableWillClose?: Function,
+type SwipeableExcludes = Exclude<
+  keyof PanGestureHandlerProps,
+  'onGestureEvent' | 'onHandlerStateChange'
+>;
+
+interface SwipeableProps
+  extends Pick<PanGestureHandlerProps, SwipeableExcludes> {
+  enableTrackpadTwoFingerGesture?: boolean;
+  friction?: number;
+  leftThreshold?: number;
+  rightThreshold?: number;
+  overshootLeft?: boolean;
+  overshootRight?: boolean;
+  overshootFriction?: number;
+  onSwipeableLeftOpen?: () => void;
+  onSwipeableRightOpen?: () => void;
+  onSwipeableOpen?: () => void;
+  onSwipeableClose?: () => void;
+  onSwipeableLeftWillOpen?: () => void;
+  onSwipeableRightWillOpen?: () => void;
+  onSwipeableWillOpen?: () => void;
+  onSwipeableWillClose?: () => void;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1]
+   * dragAnimatedValue: [0, +]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
   renderLeftActions?: (
-    progressAnimatedValue: any,
-    dragAnimatedValue: any
-  ) => any,
+    progressAnimatedValue: Animated.AnimatedInterpolation,
+    dragAnimatedValue: Animated.AnimatedInterpolation
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1]
+   * dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
   renderRightActions?: (
-    progressAnimatedValue: any,
-    dragAnimatedValue: any
-  ) => any,
-  useNativeAnimations: boolean,
-  animationOptions?: Object,
-  containerStyle?: Object,
-  childrenContainerStyle?: Object,
-};
+    progressAnimatedValue: Animated.AnimatedInterpolation,
+    dragAnimatedValue: Animated.AnimatedInterpolation
+  ) => React.ReactNode;
+  useNativeAnimations?: boolean;
+  animationOptions?: Record<string, unknown>;
+  containerStyle?: StyleProp<ViewStyle>;
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+}
+
 type StateType = {
-  dragX: Animated.Value,
-  rowTranslation: Animated.Value,
-  rowState: number,
-  leftWidth: number | typeof undefined,
-  rightOffset: number | typeof undefined,
-  rowWidth: number | typeof undefined,
+  dragX: Animated.Value;
+  rowTranslation: Animated.Value;
+  rowState: number;
+  leftWidth: number | typeof undefined;
```

Comment:
Do you know why there is `typeof undefined` instead od `?` at the beginning?

Reply:
This is coming from [this PR](https://github.com/software-mansion/react-native-gesture-handler/pull/1280/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&file-filters%5B%5D=.yml&file-filters%5B%5D=dotfile#diff-b6b2585b3e3bdb42c169114a9b93f81f84cbfefd6451620ea5fb175abd6f25fcR51) and looks like this is Flow syntax which is accidentally compatible with TS. Good catch!

Reply:
Fixed in e1c25ef.